rust/crd/src/lib.rs

```diff
@@ -160,28 +162,89 @@ impl SparkApplication {
     }
 
     pub fn volumes(&self) -> Vec<Volume> {
-        let tmp = self.spec.volumes.as_ref();
-        tmp.iter().flat_map(|v| v.iter()).cloned().collect()
+        let mut result: Vec<Volume> = self
+            .spec
+            .volumes
+            .as_ref()
+            .iter()
+            .flat_map(|v| v.iter())
+            .cloned()
+            .collect();
+
+        if self.spec.image.is_some() {
+            result.push(
+                VolumeBuilder::new(VOLUME_MOUNT_NAME_JOB)
+                    .empty_dir(EmptyDirVolumeSource::default())
+                    .build(),
+            );
+        }
+
+        if self.requirements().is_some() {
+            result.push(
+                VolumeBuilder::new(VOLUME_MOUNT_NAME_REQ)
+                    .empty_dir(EmptyDirVolumeSource::default())
+                    .build(),
+            );
+        }
```

Comment:
What are these for?

Reply:
The first one is used by an init container to copy any job jarfiles from a user provided image to the actual spark container. The second is similar but for with Python requrements.